sharding/collator/collator_test.go

```diff
@@ -89,3 +99,32 @@ func TestIsAccountInCollatorPool(t *testing.T) {
 		t.Fatal("Account not in collator pool when expected to be")
 	}
 }
+
+func TestJoinCollatorPool(t *testing.T) {
+	backend, smc := setup()
```

Comment:
How extensible is this approach to future testing? Do we have to initialize it like this within each testing method?

Reply:
Yeah, I think so. I added a TODO to consider moving this setup to some common testing package since the proposer would likely want the same setup methods and mockClient.